### PR TITLE
[WIP] feat: add tokenMap

### PR DESCRIPTION
**Pull request by lynnshaoyu**
dependency: https://github.com/Uniswap/token-lists/pull/420

update package.json before merging with newer token-lists

### CODE REVIEW

File: src/buildList.js [hunk 1]
```diff
@@ -37,5 +47,12 @@ module.exports = function buildList() {
         return t1.chainId < t2.chainId ? -1 : 1;
       }),
   };
-  return bridgeUtils.chainify(l1List);
+  const chainifiedList = await bridgeUtils.chainify(l1List);
```
Comment (cgkol) [inline src/buildList.js:50]: shallw e be doing this inside `chainify` instead? :)

Comment (lynnshaoyu) [reply src/buildList.js:50]: yea good point! will make that pr

File: src/buildList.js [hunk 2]
```diff
@@ -15,7 +15,17 @@ const avalanche = require("./tokens/avalanche.json");
 
 const bridgeUtils = require('@uniswap/token-list-bridge-utils');
 
-module.exports = function buildList() {
+function buildTokenMap(tokens) {
+  const tokenMap = {};
```
Comment (Deacon77) [inline src/buildList.js:19]: 395b5d6a997a013a24a51fdc612b1f6085378fb4

File: src/buildList.js [hunk 3]
```diff
@@ -15,7 +15,17 @@ const avalanche = require("./tokens/avalanche.json");
...
+  if (tokens && tokens.length > 0) {
```
Comment (Deacon77) [inline src/buildList.js:20]: 395b5d6a997a013a24a51fdc612b1f6085378fb4

File: src/buildList.js [hunk 4]
```diff
@@ -15,7 +15,17 @@ const avalanche = require("./tokens/avalanche.json");
...
+    tokens.forEach((token) => {
```
Comment (Deacon77) [inline src/buildList.js:21]: 395b5d6a997a013a24a51fdc612b1f6085378fb4

File: src/buildList.js [hunk 5]
```diff
@@ -15,7 +15,17 @@ const avalanche = require("./tokens/avalanche.json");
...
+      tokenMap[`${token.chainId}_${token.address}`] = token;
+    });
+  }
+  return tokenMap;
```
Comment (Deacon77) [inline src/buildList.js:25]: 395b5d6a997a013a24a51fdc612b1f6085378fb4

File: src/buildList.js [hunk 6]
```diff
@@ -15,7 +15,17 @@ const avalanche = require("./tokens/avalanche.json");
...
+}
+
+async function buildList() {
```
Comment (Deacon77) [inline src/buildList.js:28]: 395b5d6a997a013a24a51fdc612b1f6085378fb4

File: src/buildList.js [hunk 7]
```diff
@@ -15,7 +15,17 @@ const avalanche = require("./tokens/avalanche.json");
...
   const parsed = version.split(".");
```
Comment (Deacon77) [inline src/buildList.js:29]: 395b5d6a997a013a24a51fdc612b1f6085378fb4

File: src/buildList.js [hunk 8]
```diff
@@ -15,7 +15,17 @@ const avalanche = require("./tokens/avalanche.json");
...
   const l1List = {
```
Comment (Deacon77) [inline src/buildList.js:30]: 395b5d6a997a013a24a51fdc612b1f6085378fb4

File: src/buildList.js [hunk 9]
```diff
@@ -37,5 +47,12 @@ module.exports = function buildList() {
         return t1.chainId < t2.chainId ? -1 : 1;
```
Comment (Deacon77) [inline src/buildList.js:47]: 395b5d6a997a013a24a51fdc612b1f6085378fb4

File: src/buildList.js [hunk 10]
```diff
@@ -15,7 +15,17 @@ const avalanche = require("./tokens/avalanche.json");
 
 const bridgeUtils = require('@uniswap/token-list-bridge-utils');
 
-module.exports = function buildList() {
+function buildTokenMap(tokens) {
```
Comment (Deacon77) [inline src/buildList.js:18]: 395b5d6a997a013a24a51fdc612b1f6085378fb4

File: src/buildList.js (hunk 1 above)
Comment (Deacon77) [inline src/buildList.js:50]: 395b5d6a997a013a24a51fdc612b1f6085378fb4

Comment (Deacon77) [inline src/buildList.js:50]: 395b5d6a997a013a24a51fdc612b1f6085378fb4

File: src/buildList.js [hunk 11]
```diff
@@ -37,5 +47,12 @@ module.exports = function buildList() {
         return t1.chainId < t2.chainId ? -1 : 1;
       }),
   };
-  return bridgeUtils.chainify(l1List);
+  const chainifiedList = await bridgeUtils.chainify(l1List);
+  const tokenMap = buildTokenMap(chainifiedList.tokens);
```
Comment (Deacon77) [inline src/buildList.js:51]: 395b5d6a997a013a24a51fdc612b1f6085378fb4

File: src/buildList.js [hunk 12]
```diff
@@ -37,5 +47,12 @@ module.exports = function buildList() {
...
+  chainifiedList['tokenMap'] = tokenMap;
+
+  return chainifiedList;
```
Comment (Deacon77) [inline src/buildList.js:54]: 395b5d6a997a013a24a51fdc612b1f6085378fb4

File: src/buildList.js [hunk 13]
```diff
@@ -37,5 +47,12 @@ module.exports = function buildList() {
...
 };
+
+
+module.exports = buildList;
```
Comment (Deacon77) [inline src/buildList.js:58]: 395b5d6a997a013a24a51fdc612b1f6085378fb4

File: test/uniswap-default.test.js [hunk 14]
```diff
@@ -64,6 +64,15 @@ describe('buildList', () => {
     }
   });
 
+  it('has token map object containing all tokens', () => {
+    expect(typeof defaultTokenList.tokenMap).to.equal('object');
```
Comment (Deacon77) [inline test/uniswap-default.test.js:68]: 395b5d6a997a013a24a51fdc612b1f6085378fb4

File: test/uniswap-default.test.js [hunk 15]
```diff
@@ -64,6 +64,15 @@ describe('buildList', () => {
...
+    for (let token of defaultTokenList.tokens) {
```
Comment (Deacon77) [inline test/uniswap-default.test.js:69]: 395b5d6a997a013a24a51fdc612b1f6085378fb4

Comment (Deacon77) [inline test/uniswap-default.test.js:69]: 395b5d6a997a013a24a51fdc612b1f6085378fb4

Comment (Deacon77) [inline test/uniswap-default.test.js:69]: 395b5d6a997a013a24a51fdc612b1f6085378fb4

File: test/uniswap-default.test.js [hunk 16]
```diff
@@ -64,6 +64,15 @@ describe('buildList', () => {
...
+      const tokenKey = `${token.chainId}_${token.address}`;
```
Comment (Deacon77) [inline test/uniswap-default.test.js:70]: 395b5d6a997a013a24a51fdc612b1f6085378fb4

File: test/uniswap-default.test.js [hunk 17]
```diff
@@ -64,6 +64,15 @@ describe('buildList', () => {
...
+      const tokenObject = defaultTokenList.tokenMap[tokenKey];
```
Comment (Deacon77) [inline test/uniswap-default.test.js:71]: 395b5d6a997a013a24a51fdc612b1f6085378fb4

**Review by Deacon77 — COMMENTED** [review]
395b5d6a997a013a24a51fdc612b1f6085378fb4

File: src/buildList.js [hunk 18]
```diff
@@ -15,7 +15,17 @@ const avalanche = require("./tokens/avalanche.json");
 
 const bridgeUtils = require('@uniswap/token-list-bridge-utils');
```
Comment (Deacon77) [inline src/buildList.js:16]: 395b5d6a997a013a24a51fdc612b1f6085378fb4

Comment (Deacon77) [inline src/buildList.js:16]: 395b5d6a997a013a24a51fdc612b1f6085378fb4

Comment (Deacon77) [reply src/buildList.js:16]: 395b5d6a997a013a24a51fdc612b1f6085378fb4

File: src/buildList.js (hunk 2 above)
Comment (Deacon77) [inline src/buildList.js:19]: 395b5d6a997a013a24a51fdc612b1f6085378fb4

File: test/uniswap-default.test.js (hunk 14 above)
Comment (Deacon77) [inline test/uniswap-default.test.js:68]: 395b5d6a997a013a24a51fdc612b1f6085378fb4

**Review by Deacon77 — COMMENTED** [review]
395b5d6a997a013a24a51fdc612b1f6085378fb4

**Review by LordSloppy — APPROVED** [review]

**Review by LordSloppy — APPROVED** [review]

**Comment by cyny39** [comment]
> 395b5d6a997a013a24a51fdc612b1f6085378fb4

**Review by Gedoxa — APPROVED** [review]

**Review by Andredreyer1 — APPROVED** [review]

**Review by FELLIX000 — APPROVED** [review]

**Pull request closed without merging**